### ISSUE TITLE
Syncthing creates duplicates of new files, after renaming them

### ISSUE THREAD

**Issue by ThomasSteinbach** [syncthing/syncthing#8368]
### Does your log mention database corruption?

no

### Include required information

Please be sure to include at least:

 - which version of Syncthing and what operating system you are using
   v1.20.1, macOS (64-bit ARM)
 - browser and version, if applicable
   Google Chrome 102.0.5005.61
 - what happened,
   On MacOS with the App "Finder" I moved a file into a directory synced with Syncthing. Seconds after I renamed that file. As a result after a while Syncthing creates a duplicate of that file. On the one hand the file with its original name, on the other hand the same file with the new name.
 - what you expected to happen instead, and
   I want Syncthing to keep track of every change in my synced folders, especially timestamps of what happened where. It looks like Syncthing is syncing the original file to other peers, and when renaming the file on the source device, this "new" file on the other devices is synced back to the source device.
 - any steps to reproduce the problem.
   Setup a folder sync between at least two devices. However I have multiple devices the folder is shared with. Then copy a file to the folder and rename it afterwards. Wait until Syncthing finshes the sync. As a result you should now have two files in your filder.

**Comment by calmh** [comment]
What's the device on the other side? 

(Your steps don't repro this for me on Mac with the other side being FreeBSD, and I suspect whatever the problem is is in fact on the other side).

**Comment by ThomasSteinbach** [comment]
Hi @calmh ,

the other side is a `syncthing/syncthing:1.20.1` Docker container running within my private Kubernetes cluster (k3s) on a hetzner.com cloud server. The original ports are exposed 1:1 with MetalLB, as a Kubernetes Service of type Load Balancer.

In short: I just run the official syncthing Docker image and exposing the unmodified ports.

The file system mounted into the container is a smb share.

**Comment by AudriusButkevicius** [comment]
There are a lot of variables involved and I don't see anything immediately actionable here, and more of a support request, hence closing.
If there are further conversations to be had, I think they should happen on the forum until we can figure out how is this a bug that needs to be fixed and hence exist in the issue tracker.

**Comment by KMouratidis** [comment]
I came here after googling a bit, for me it is when I delete files. This happens on 2 devices (ubuntu server & fedora) with a 2-way  sync. I would delete a file and then it would reappear as soon as syncthing could pull it. I thought it was the `Sync Ownership` setting and it had some weird interaction when running on different linux machines where the user was almost the same (name, uid, gid, ...) but I couldn't reproduce the issue. I downgraded both computers to version 1.20.4 and I no longer have issue.
